lib/ftp_session.dart

```diff
@@ -77,10 +110,10 @@ class FtpSession {
     int port = dataListener!.port;
     int p1 = port >> 8;
     int p2 = port & 0xFF;
-    String address = controlSocket.address.address.replaceAll('.', ',');
-
+    // String address = controlSocket.address.address.replaceAll('.', ',');
     // I'm not sure what happened here, the client shows nothing if I comment out this line.
-    await Future.delayed(const Duration(microseconds: 100));
+    // await Future.delayed(const Duration(microseconds: 0));
+    var address = (await _getIpAddress()).replaceAll('.', ',');
     sendResponse('227 Entering Passive Mode ($address,$p1,$p2)');
     dataListener!.first.then((socket) {
       dataSocket = socket;
```

Comment:
i think having the `sendResponse('150 Opening data connection');` here is the better apporach since all commands that needs it looks like it wants that response.

Reply:
i meant after the `dataSocket = socket;` line